eisbuk-admin/src/components/atoms/IntervalCard/__tests__/IntervalCard.test.tsx

```diff
@@ -1,16 +1,75 @@
 import React from "react";
-import { render } from "@testing-library/react";
+import {cleanup, screen render } from "@testing-library/react";
 
 import IntervalCard from "../IntervalCard";
+import * as bookingActions from "@/store/actions/bookingOperations";
+import { bookedSlot, nonBookedSlot } from "../__testData__/dummyData";
 
-import { baseProps } from "../__testData__/dummyData";
+import { Customer, Slot } from "eisbuk-shared";
+import { SlotInterface, SlotInterval } from "@/types/temp";
+
+
+const mockDispatch = jest.fn();
+
+jest.mock("react-redux", () => ({
+  ...jest.requireActual("react-redux"),
+  useDispatch: () => mockDispatch,
+  /** We're mocking this not to fail tests including SlotForm (this will be returning newSlotTime as null) @TODO remove in the future */
+  useSelector: () => null,
+}));
+
+jest.mock("react-router-dom", () => ({
+  ...jest.requireActual("react-router-dom"),
+  useParams: () => ({ secretKey: "secret_key" }),
+}));
```

Comment:
You don't need this as you're not relying on the router anywhere within the component, nor do you need `secretKey` anywhere.